crates/ratchet-core/src/tensor.rs

```diff
@@ -838,6 +839,17 @@ impl Tensor {
         ))
     }
 
+    #[cfg(feature = "testing")]
+    pub fn to_tch<T: TensorDType + tch::kind::Element>(&self) -> anyhow::Result<tch::Tensor> {
```

Comment:
Should we take `self` and consume here instead of `deep_clone`?

Reply:
We usually pass a `&Tensor` in the tests to get the ground truth so I don't think we can consume `self` here. Although we *may* not need the  `deep_clone` as we call already call `into_owned` which has the signature : 
```rust
    pub fn into_owned(self) -> Array<A, D>
    where
        A: Clone,
        S: Data,
    {
        S::into_owned(self)
    }
``` 

So we are cloning elements either way.